lib/msf/core/opt_enum.rb

```diff
@@ -1,48 +1,45 @@
 # -*- coding: binary -*-
 
 module Msf
+  ###
+  #
+  # Enum option.
+  #
+  ###
+  class OptEnum < OptBase
+    def type
+      return 'enum'
+    end
 
-###
-#
-# Enum option.
-#
-###
-class OptEnum < OptBase
-
-  def type
-    return 'enum'
-  end
-
-  def valid?(value=self.value, check_empty: true)
-    return false if check_empty && empty_required_value?(value)
-    return true if value.nil? and !required?
-
-    (value and self.enums.include?(value.to_s))
-  end
-
-  def normalize(value=self.value)
-    return nil if not self.valid?(value)
-    return value.to_s
-  end
-
-  def desc=(value)
-    self.desc_string = value
+    # This overrides required default from 'false' to 'true'
+    def initialize(in_name, attrs = [],
+                   required: true, desc: nil, default: nil, enums: [], aliases: [])
+      super
+    end
 
-    self.desc
-  end
+    def valid?(value = self.value, check_empty: true)
+      return false if check_empty && empty_required_value?(value)
+      return true if value.nil? && !required?
 
-  def desc
-    if self.enums
-      str = self.enums.join(', ')
+      !value.nil? && enums.include?(value.to_s)
     end
-    "#{self.desc_string || ''} (Accepted: #{str})"
-  end
 
+    def normalize(value = self.value)
+      !valid?(value) ? nil : value.to_s
```

Comment:
"You're my ternary hero" - show that to Egypt, watch him turn purple.

Reply:
Funny enough, the style guide now endorses ternary. ;)

Reply:
To quote the bug from family guy who started fires - "good, good."

Reply:
life is too short, let's make cool stuff

Reply:
Hey I'm all for this, its readable and neat. If the method name was not explicit, this might suck, but as is it reads out loud in plain English.

Reply:
Could we at least get rid of the unnecessary `!` and leave it as `valid?(value) ? value.to_s : nil`?

Reply:
/me turns purple